Correct SessionWrapper unit tests

rclee opened the pull request:
Some of the tests didn't use proper tooling and were all in all a bit messy. 

This commit should clean up the unit tests around the session wrapper and stubs the `handlechallenge` method to ensure that it returns the correct value for unit testing. The test we had before (testing `NSURLSessionAuthChallengeCancelAuthenticationChallenge` may be added back as an integration test). 

LLVM_LTO is also disabled for debug modes. This is to allow breakpoints to operate properly. `-fcxx-modules` and `-fmodules` are added to import #TrustKit into the .mm file.

Co-authored-by: Andrew-Kitchen <andrew.kitchen@mapbox.com>

rclee committed:
Correct SessionWrapper unit tests

Some of the tests didn't use proper tooling and were all in all a bit messy. 

This commit should clean up the unit tests around the session wrapper and stubs the `handlechallenge` method to ensure that it returns the correct value for unit testing. The test we had before (testing `NSURLSessionAuthChallengeCancelAuthenticationChallenge` may be added back as an integration test). 

LLVM_LTO is also disabled for debug modes. This is to allow breakpoints to operate properly. `-fcxx-modules` and `-fmodules` are added to import #TrustKit into the .mm file.

Co-authored-by: Andrew-Kitchen <andrew.kitchen@mapbox.com>

rclee committed:
Correct SessionWrapper unit tests

Some of the tests didn't use proper tooling and were all in all a bit messy. 

This commit should clean up the unit tests around the session wrapper and stubs the `handlechallenge` method to ensure that it returns the correct value for unit testing. The test we had before (testing `NSURLSessionAuthChallengeCancelAuthenticationChallenge` may be added back as an integration test). 

LLVM_LTO is also disabled for debug modes. This is to allow breakpoints to operate properly. `-fcxx-modules` and `-fmodules` are added to import #TrustKit into the .mm file.

Co-authored-by: Andrew-Kitchen <andrew.kitchen@mapbox.com>

rclee committed:
Merge branch 'master' into rclee-sessionwrapper-unit-tests

akitchen commented:
The `context` at line 78 and line 147 seem to be the same thing, since the tests themselves run synchronously on the main queue. Not sure if you need to keep both of these around. Other than that, LGTM :)

rclee committed:
Remove duplicate test

The removed context is already covered by the context at line 78 since the tests themselves run synchronously on the main queue.

rclee committed:
Remove duplicate test

The removed context is already covered by the context at line 78 since the tests themselves run synchronously on the main queue.

rclee commented on MapboxMobileEvents.xcodeproj/project.pbxproj:
```diff
@@ -1171,6 +1171,7 @@
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
 				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				LLVM_LTO = NO;
```
Yeah, I needed to turn off LTO on debug builds -- they were preventing breakpoints from operating for some reason.

rclee committed:
Merge branch 'master' into rclee-sessionwrapper-unit-tests

rclee committed:
Fix hasty test removal

rclee committed:
Fix hasty test removal

rclee committed:
Stop setting loggerBlock for TrustKit

We're not making use of the TrustKit logger and it's causing issues with Maps nightly builds. I believe removing this should fix that.

rclee committed:
Stop setting loggerBlock for TrustKit

We're not making use of the TrustKit logger and it's causing issues with Maps nightly builds. I believe removing this should fix that.

rclee commented:
Implemented your suggestions and removed the setting of the TrustKit Logger which is only used in debug mode. The Logger was removed to fix an error on nightly builds (we're not using the Logger anyway).

alfwatt reviewed (dismissed):
Changes look good, make sure to rebase the branch, the `LLVM_LTO` change might conflict (or be overrides by) changes on `master` which disabled it for all targets in the project

rclee committed:
telemetryMetrics improvements (#113)

* Reformat failedRequests metric 

This should allow the failedRequests metric to be more queryable.

* Track received data 

Implement the event schema changes to begin tracking received data on all requests. 

Tested using breakpoints to track each `data.length` from start to finish.

* Fix bug in event counting

Although this wouldn't be much an issue in our current implementation; once we add in deferred telemetry we would have missed counting any additional batches in APIClient.

Also adds a failed event count if the request is ever nil.

* Implement suggested changes 

Also adds missing tests and removed the synchronous NSURLConnection call that was being used for testing. 

Changed transferred bytes to sent bytes -- this will be reflected in the schema soon.

* Move telemetryMetrics send request method

Moves `telemetryMetrics` send request method to `flush` to have the event send more regularly. 

Previously this method was moved out of `flush` to be more readable but we missed a vital part that `flush` is also called on a timer and `sendTelemetryMetricsEvent` was out of that loop.

* Add telemetryMetrics timer test

Adds a unit test to ensure that we attempt to send a telemetryMetrics event when the event queue timer fires.

rclee committed:
Merge remote-tracking branch 'origin/rclee-sessionwrapper-unit-tests' into rclee-sessionwrapper-unit-tests
